pkg/printers/internalversion/printers.go

```diff
@@ -2181,8 +2181,12 @@ func layoutContainerCells(containers []api.Container) (names string, images stri
 	return namesBuffer.String(), imagesBuffer.String()
 }
 
-// formatEventSource formats EventSource as a comma separated string excluding Host when empty
-func formatEventSource(es api.EventSource) string {
+// formatEventSource formats EventSource as a comma separated string excluding Host when empty,
+// or returns reportingController when EventSource is empty
+func formatEventSource(es api.EventSource, reportingController string) string {
```

Comment:
IIUC, source.{component,host} is deprecated in favor of reportingController/reportingInstance

so I think this should look more like:

```go
func formatEventSource(es api.EventSource, reportingController, reportingInstance string) string {
  return formatEventSourceComponentInstance(
    firstNonEmpty(es.Component, reportingController),
    firstNonEmpty(es.Host, reportingInstance),
  )
}

func firstNonEmpty(ss ...string) string {
  for _, s := range ss {
    if len(s) > 0 {
      return s
    }
  }
  return ""
}

func formatEventSourceComponentInstance(component, instance string) string {
  if len(instance) == 0 {
    return component
  }
  return component + ", " + instance
}
```

Reply:
Make sense. Done.